### ISSUE TITLE
Indexes of the counted cycle range

### ISSUE THREAD

**Issue by addarnr** [iamlikeme/rainflow#31]
Hello,

I am in search of a rainflow counting algorithm in Python that provides the indexes of the counted ranges (similar to the rainflow counting in Matlab) also.  Would this implementation rainflow() (rainflow 2.2.0 by PyPI) provide the indexes? If not, is there a way to modify this rainflow() code to obtain the indexes?
Are there any other Python based rainflow implementations that provide indexes? 
Thank you!

**Comment by iamlikeme** [comment]
Hi! The current version does not provide indices, but it seems fairly straightforward to implement. Are you willing to implement this feature and submit a pull request? Or at least a test case?
I am not sure about other Python based implementation of rainflow counting. You would have to research yourself.

**Comment by addarnr** [comment]
Hello,
Yes, I am willing to test a case.  
Thanks you!

**Comment by addarnr** [comment]
Hello, please let me know what to do at my end.
Thank you!

**Comment by iamlikeme** [comment]
Hi again. It would be great if you could provide a sample time series and the expected result including indices. Some sample time series are already included in the unit tests, e.g. [this one](https://github.com/iamlikeme/rainflow/blob/b1990b4e0c9189479df0b5f720c842e8a9bc66a1/tests/test_rainflow.py#L6). Perhaps you could generate the expected result using Matlab?

**Comment by addarnr** [comment]
Hello, yes i will do that and get back to you.

Thanks!

**Comment by addarnr** [comment]
Hello,
I tested (2) series, one from the link you sent me and one my own.  The resulting cycle counts, ranges, means, start indices,  and end indices are attached.

Thank you!

Sent from Mail for Windows 10

**Comment by iamlikeme** [comment]
Thank you for the test cases. I will work on an implementation and get back to you.

**Comment by gsokoll** [comment]
What is the benefit of this functionality ?  Other than for academic interest, it seems to have little to no real world use.

**Comment by addarnr** [comment]
No, that is not true.  
There are cases where the rainflow count of one signal can be used as a representation of another signal that cannot be directly measured.  Throttling of a car is an example.  Every throttle change causes some damage to the car engine.  You can estimate these damages by the rainflow count of the throttle if there is data already available to correlate the two.  In this case you need the indices of the throttle rainflow counts to correlate the two.

**Comment by gsokoll** [comment]
Note that the Matlab rainflow implementation includes an option to return the indices of reversals identified in the data, not cycles.  This is not the same as what you have asked for here.
https://www.mathworks.com/help/signal/ref/rainflow.html

**Comment by addarnr** [comment]
Yes the linear indices are what I am after. Sorry if i was not clear. The
tests i performed include the

**Comment by iamlikeme** [comment]
The feature you requested is now released in v3.0.0 on PyPI. Here is a usage example:

```python
series = [0, -2, 1, -3, 5, -1, 3, -4, 4, -2, 0]
list(rainflow.extract_cycles(series))
# Output
[(2, -1.0, 0.5, 0, 1),
(3, -0.5, 0.5, 1, 2),
(4, -1.0, 0.5, 2, 3),
...
(2, -1.0, 0.5, 9, 10)]
```
